theano/compile/builders.py

```diff
@@ -251,7 +284,18 @@ def __init__(
         self.kwargs = kwargs
         self.input_types = [inp.type for inp in inputs]
         self.output_types = [out.type for out in outputs]
-        self.set_grad_overrides(grad_overrides)
+        cond = int(lop_overrides == 'default') * 2 + int(grad_overrides == 'default')
+        if cond == 0:
+            raise ValueError('lop_overrides and grad_overrides are mutually exclusive')
+        elif cond == 1:
+            self.set_lop_overrides(lop_overrides)
+            self._lop_type = 'lop'
+        elif cond == 2:
+            self.set_lop_overrides(grad_overrides)
+            self._lop_type = 'grad'
+        else:
+            self.set_lop_overrides('default')
+            self._lop_type = 'lop'
```

Comment:
This trickery is confusing to read (even if correct).  Please use normal if/else for this.